### PR TITLE
Testing

### PR DESCRIPTION
**Pull request by sebytremblay**
Implements baseline testing for model, controller, and router. Also implements small fixes to observed issues from tests. One test is failing and has been logged on Linear.

### CODE REVIEW

File: test/controller/beast.test.ts [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+jest.mock('../../src/models/beast');
+
+const mockResponse = () => {
+    const res = {} as Response;
+    res.status = jest.fn().mockReturnValue(res);
+    res.json = jest.fn().mockReturnValue(res);
+    return res;
+};
+
+const mockRequest = (body: Record<string, any> = {}, params: Record<string, any> = {}) => ({
```
Comment (cdugo) [inline test/controller/beast.test.ts:15]: Can we move the mockRequest & response into a test helper file? I feel like we can reuse these across multiple test files

File: test/controller/beast.test.ts [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+
+            expect(res.status).toHaveBeenCalledWith(201);
+            expect(res.json).toHaveBeenCalledWith(expect.objectContaining({ message: 'Successfully created' }));
+        });
+
+        it('handles CustomError when creation fails', async () => {
+            const req = mockRequest({ gamerTag: 'TestTag', email: 'test@example.com' });
+            const res = mockResponse();
+
+            (BeastModel.create as jest.Mock).mockRejectedValue(new BadRequestError('Creation failed'));
```
Comment (cdugo) [inline test/controller/beast.test.ts:45]: Why is this failing? Also do you think it'd be beneficial to have more specific error messages instead of 'Creation failed'? 

It might be useful to be more descriptive in the 'it' statements - why should creation fail?

Comment (cdugo) [reply test/controller/beast.test.ts:45]: I think there are more potential edge cases we should test for (like creating a user with a username that already exists, etc)

Comment (sebytremblay) [reply test/controller/beast.test.ts:45]: There is no particular reason it fails. We are forcing the model to error out via the mocks to verify the controller handles errors properly and returns a 400 Bad Status. I guess we could do with different types of failures beyond BadRequestError if that's what you mean.
